src/test/java/seedu/address/model/expense/UniqueExpenseListTest.java

```diff
@@ -1,4 +1,175 @@
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_DESCRIPTION_BUILD_A_BEAR;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_EXPENSE_CATEGORY_TRANSPORT;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalExpenses.ANNUAL_NETFLIX_FEES;
+import static seedu.address.testutil.TypicalExpenses.ANNUAL_SPOTIFY;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.expense.exceptions.DuplicateExpenseException;
+import seedu.address.model.expense.exceptions.ExpenseNotFoundException;
+import seedu.address.testutil.ExpenseBuilder;
+
 public class UniqueExpenseListTest {
+
+    private final UniqueExpenseList UniqueExpenseList = new UniqueExpenseList();
+
+    @Test
+    public void contains_nullExpense_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> UniqueExpenseList.contains(null));
+    }
+
+    @Test
+    public void contains_expenseNotInList_returnsFalse() {
+        assertFalse(UniqueExpenseList.contains(ANNUAL_SPOTIFY));
+    }
+
+    @Test
+    public void contains_expenseInList_returnsTrue() {
+        UniqueExpenseList.add(ANNUAL_SPOTIFY);
+        assertTrue(UniqueExpenseList.contains(ANNUAL_SPOTIFY));
+    }
+
+    // Unique identity fields: all fields but ExpenseCategory
+    @Test
+    public void contains_expenseWithSameIdentityFieldsInList_returnsTrue() {
+        UniqueExpenseList.add(ANNUAL_SPOTIFY);
+        Expense editedSpotify = new ExpenseBuilder(ANNUAL_SPOTIFY).withExpenseCategory(VALID_EXPENSE_CATEGORY_TRANSPORT)
+                .build();
+        assertTrue(UniqueExpenseList.contains(editedSpotify));
+    }
+
+    @Test
+    public void add_nullExpense_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> UniqueExpenseList.add(null));
+    }
+
+    @Test
+    public void add_duplicateExpense_throwsDuplicateExpenseException() {
+        UniqueExpenseList.add(ANNUAL_SPOTIFY);
+        assertThrows(DuplicateExpenseException.class, () -> UniqueExpenseList.add(ANNUAL_SPOTIFY));
+    }
+
+    @Test
+    public void setExpense_nullTargetExpense_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> UniqueExpenseList.setExpense(null, ANNUAL_SPOTIFY));
+    }
+
+    @Test
+    public void setExpense_nullEditedExpense_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> UniqueExpenseList
+                .setExpense(ANNUAL_SPOTIFY, null));
+    }
+
+    @Test
+    public void setExpense_targetExpenseNotInList_throwsExpenseNotFoundException() {
+        assertThrows(ExpenseNotFoundException.class, () -> UniqueExpenseList
+                .setExpense(ANNUAL_SPOTIFY, ANNUAL_SPOTIFY));
```

Comment:
Maybe should use 2 different expenses in the illustration to show a clearer picture, because this can be a little confusing for the example, can consider using BUILD_A_BEAR.

Reply:
i think its fine this way... this was the way which was implemented in the skeleton as well